File: app/schemas/membership.py
from datetime import datetime

from pydantic import BaseModel

from app.models import MembershipRole, MembershipStatus


class MembershipResponse(BaseModel):
    id: str
    organization_id: str
    user_id: str
    role: MembershipRole
    status: MembershipStatus
    created_at: datetime
    accepted_at: datetime | None


class MembershipRoleUpdate(BaseModel):
    role: MembershipRole

File: frontend/src/types/membership.ts
export type MembershipRole = "PLATFORM_ADMIN" | "ORGANIZATION_ADMIN" | "MANAGER" | "AGENT" | "VIEWER";
export type MembershipStatus = "INVITED" | "ACTIVE" | "SUSPENDED" | "ARCHIVED";

export interface Membership {
    id: string;
    organization_id: string;
    user_id: string;
    role: MembershipRole;
    status: MembershipStatus;
    created_at: string;
    accepted_at: string | null;
}

export interface InvitationResponse {
    id: string;
    email: string;
    role: MembershipRole;
    expires_at: string;
    invitation_token: string;
}
